### ISSUE TITLE
Use `std::time::Duration` for time intervals

### ISSUE THREAD

**Issue by Herschel** [ruffle-rs/ruffle#6075]
Time durations in Ruffle are often stored as floats ([example](https://github.com/ruffle-rs/ruffle/blob/master/core/src/player.rs#L570-L582)), and it's difficult to know which units the values are in (secs, ms, ...).

It'd be nice to switch these over to use [`std::time::Duration`](https://doc.rust-lang.org/stable/std/time/struct.Duration.html) for clarity.

**Comment by Udai1931** [comment]
Hi @Herschel I like to work on this.

**Comment by Mekomancer** [comment]
@Udai1931 you still one this, i like to have a knack at it #6151

**Comment by stevenhuyn** [comment]
Are you like suggesting every place where durations are used should be switched out to durations?
[Example](https://github.com/ruffle-rs/ruffle/blob/a320e3b1d0d4f271aa1f079574a969873e5967aa/core/src/player.rs#L563)

Or more like all functions input and return types should be `std::time::Duration`. Also is this largely limited to this file or there would be a lot of place?

**Comment by stevenhuyn** [comment]
Are you like suggesting every place where durations are used should be switched out to durations?
[Example](https://github.com/ruffle-rs/ruffle/blob/a320e3b1d0d4f271aa1f079574a969873e5967aa/core/src/player.rs#L563)

Or more like all functions input and return types should be `std::time::Duration`. Also is this largely limited to this file or there would be a lot of place?

**Comment by higumachan** [comment]
@Herschel 
This, issue has been without activity for a long time, can I try to work on it?

**Comment by higumachan** [comment]
@n0samu @Herschel 

I tried use `std::time::Duration`

However, as noted in this PR, the `cargo:test://regression_tests::as3_sound_embeddedprops` test case fails.
https://github.com/higumachan/ruffle/pull/1

One possible reason for this is that the original floating-point data was used to handle the data, so when division is performed, values finer than microseconds are being generated.

My solution, as I see it, is as follows.

1. create an original Duration type that is compatible for ruffle
2. discard compatibility

Which do you think is better, or do you have any other good ideas?

**Comment by Herschel** [comment]
`std::time::Duration` only uses nanosecond resolution (~10 significant figures), so it loses some precision compared to calculating the sound sample rate directly (~16 sig figures in `f64`).

I was only really thinking about `dt` and the timers when I wrote this issue and didn't really consider the audio position -- sorry about that!

Some thoughts on the best way to proceed:
 * Seems like the easiest path forward would be to make our own `struct Duration(f64)` type.
 * This also avoids the need for `SignedDuration` (we'll just always be signed).
 * We can add `Duration::as_secs` etc. methods directly onto this type.
  * Having `duration_helper` as a separate crate is overkill; `Duration` could live in its own module in `ruffle_core`.
  
Basically the idea is to make things clearer+easier; if it gets more complicated, then it's more trouble than it's worth and we're better off sticking with `f64`.

Thanks for your work! Sorry for the delay in responding.

**Comment by higumachan** [comment]
@Herschel 

Thank you for response!

OK, I would like to implement the policy of adding `Duration(f64)` in `ruffle_core`.
